pkg/tasktesting/taskrun.go

```diff
@@ -147,6 +152,48 @@ func WaitForTaskRunDone(
 	}
 }
 
-func Done(tr *tekton.TaskRun) bool {
-	return tr.IsDone()
+func waitForTaskRunPod(
+	t *testing.T,
+	c *kubernetes.Clientset,
+	taskRunName,
+	namespace string,
+	errs chan error,
+	taskRunDone chan bool,
+	podAdded chan *v1.Pod) {
+	log.Printf("Waiting for pod related to TaskRun %s to be added to the cluster\n", taskRunName)
+	stop := make(chan struct{})
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(c, time.Second*30)
+	podsInformer := kubeInformerFactory.Core().V1().Pods().Informer()
+
+	var taskRunPod *v1.Pod
+
+	podsInformer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				// when a new task is created, watch its events
+				pod := obj.(*v1.Pod)
+				if strings.HasPrefix(pod.Name, taskRunName) {
+					taskRunPod = pod
+					log.Printf("TaskRun %s added pod %s to the cluster", taskRunName, pod.Name)
+					stop <- struct{}{}
```

Comment:
Why not doing `podAdded <- taskRunPod` directly at this point?

Reply:
I need to return from the outer function to stop that goroutine ...